presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/PrestoThriftService.java

```diff
@@ -0,0 +1,100 @@
+ */
+@ThriftService
+public interface PrestoThriftService
+        extends Closeable
+{
+    /**
+     * Returns available schema names.
+     */
+    @ThriftMethod("prestoListSchemaNames")
+    List<String> listSchemaNames()
+            throws PrestoThriftServiceException;
+
+    /**
+     * Returns tables for the given schema name.
+     *
+     * @param schemaNameOrNull a structure containing schema name or {@literal null}
+     * @return a list of table names with corresponding schemas. If schema name is null then returns
+     * a list of tables for all schemas. Returns an empty list if a schema does not exist
+     */
+    @ThriftMethod("prestoListTables")
+    List<PrestoThriftSchemaTableName> listTables(
+            @ThriftField(name = "schemaNameOrNull") PrestoThriftNullableSchemaName schemaNameOrNull)
+            throws PrestoThriftServiceException;
+
+    /**
+     * Returns metadata for a given table.
+     *
+     * @param schemaTableName schema and table name
+     * @return metadata for a given table, or a {@literal null} value inside if it does not exist
+     */
+    @ThriftMethod("prestoGetTableMetadata")
+    PrestoThriftNullableTableMetadata getTableMetadata(
+            @ThriftField(name = "schemaTableName") PrestoThriftSchemaTableName schemaTableName)
+            throws PrestoThriftServiceException;
+
+    /**
+     * Returns a batch of splits.
+     *
+     * @param schemaTableName schema and table name
+     * @param desiredColumns a superset of columns to return; empty set means "no columns", {@literal null} set means "all columns"
+     * @param outputConstraint constraint on the returned data
+     * @param maxSplitCount maximum number of splits to return
+     * @param nextToken token from a previous split batch or {@literal null} if it is the first call
+     * @return a batch of splits
+     */
+    @ThriftMethod("prestoGetSplits")
+    ListenableFuture<PrestoThriftSplitBatch> getSplits(
+            @ThriftField(name = "schemaTableName") PrestoThriftSchemaTableName schemaTableName,
+            @ThriftField(name = "desiredColumns") PrestoThriftNullableColumnSet desiredColumns,
+            @ThriftField(name = "outputConstraint") PrestoThriftTupleDomain outputConstraint,
+            @ThriftField(name = "maxSplitCount") int maxSplitCount,
+            @ThriftField(name = "nextToken") PrestoThriftNullableToken nextToken)
+            throws PrestoThriftServiceException;
+
+    /**
+     * Returns a batch of rows for the given split.
+     *
+     * @param splitId split id as returned in split batch
+     * @param columns a list of column names to return
+     * @param maxBytes maximum size of returned data in bytes
+     * @param nextToken token from a previous batch or {@literal null} if it is the first call
+     * @return a batch of table data
+     */
+    @ThriftMethod("prestoGetRows")
+    ListenableFuture<PrestoThriftPage> getRows(
```

Comment:
Hmm.. `getRows` might be more clear to the people not very familiar with Presto.